src/ClientData/include/ClientData/DataManager.h

```diff
@@ -166,6 +182,8 @@ class DataManager final {
   bool collect_memory_info_ = false;
   uint64_t memory_sampling_period_ms_ = 10;
   uint64_t memory_warning_threshold_kb_ = 8ULL * 1024 * 1024;
+
+  std::optional<TimeRange> selection_time_range_;
```

Comment:
Nit. 

I am a fan of `std::optional` and I really don't like default values, but I think, it's not the case here. Might be missing smth though. 

Can't we add a default ctor to `TimeRange` (or just have a method like `CreateAllIncludingTimeRange`) that would instantiate `TimeRange{0, UINT64_MAX}` and use this value instead of the empty optional? This should simplify the code in `OrbitApp::IsTimerActive`. And if IIRC, this is more or less the solution you had in the initial commit.

Reply:
The optional allows me to check whether or not a time range is actually in existence.  In the next PR, I will be able to select any combination of 1 thread + a time range and without the optional, I'll never know when we're going back to the whole capture.

Reply:
Cool.